src/buffer.c

```diff
@@ -2206,6 +2206,9 @@ buflist_new(
     if ((flags & BLN_CURBUF) && curbuf_reusable())
     {
 	buf = curbuf;
+	// b:undo_ftplugin may be set, undo it
+	do_cmdline_cmd((char_u*)"if exists('b:undo_ftplugin') | :legacy :exe \
+		b:undo_ftplugin | endif");
```

Comment:
Is it possible that `b:undo_ftplugin` may free the buffer?

Reply:
I locked the buffer before executing `b:undo_ftplugin`